core/commands/storage/upload/upload/init.go

```diff
@@ -158,6 +163,18 @@ the shard and replies back to client for the next challenge step.`,
 				if err != nil {
 					return err
 				}
+
+				// swarm connect to renter and challenge-node
+				for i := 0; i < 3; i++ {
+					ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+					go func() {
+						ctxParams.Api.Swarm().Connect(ctx, peer.AddrInfo{ID: requestPid})
```

Comment:
this should be done in host download shard routine, inside each backoff try.

Reply:
didn't see this connect thing.
looks like we getAllCIDsRecursive directly.